contracts/test/GrantRound.test.ts

```diff
@@ -0,0 +1,153 @@
+    registry = <GrantRegistry>await deployContract(deployer, grantRegistryArtifact);
+
+    // Create an example grant
+    await registry.connect(deployer).createGrant(deployer.address, grantPayee.address, mockUrl);
+
+    const grantRoundArtifact: Artifact = await artifacts.readArtifact('GrantRound');
+    const roundContractFactory = new ethers.ContractFactory(
+      grantRoundArtifact.abi,
+      grantRoundArtifact.bytecode,
+      deployer
+    );
+
+    roundContract = await roundContractFactory.deploy(
+      grantRoundOwner.address,
+      registry.address,
+      mockERC20.address,
+      startTime,
+      endTime,
+      mockUrl,
+      minContribution
+    );
+  });
+
+  describe('Initialization', () => {
+    it('deploys properly', async function () {
+      expect(isAddress(roundContract.address), 'Failed to deploy GrantRegistry').to.be.true;
+    });
+  });
+
+  describe('acceptMatchingPool - Add funds to matching round', () => {
+    it('updates contract and user balances', async function () {
+      // set up mock calls
+      await mockERC20.mock.balanceOf.withArgs(mpUser.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.balanceOf.withArgs(roundContract.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.transferFrom.withArgs(mpUser.address, roundContract.address, balances[0]).returns(true);
+
+      await roundContract.connect(mpUser).acceptMatchingPool(balances[0]);
+      expect(await roundContract.connect(deployer).checkBalance(roundContract.address)).to.be.equal(
+        ethers.utils.parseEther(balances[0])
+      );
+    });
+  });
+
+  describe('donateToGrant - Add funds during active round', () => {
+    it('sends donation token to the grant payee with a given grant id', async function () {
+      const grantId: number = 0;
+      await mockERC20.mock.balanceOf.withArgs(donor.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.balanceOf.withArgs(grantPayee.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.transfer.withArgs(grantPayee.address, balances[0]).returns(true);
+
+      await roundContract.connect(donor).donateToGrant(balances[0], grantId);
+      expect(await roundContract.connect(deployer).checkBalance(grantPayee.address)).to.be.equal(
+        ethers.utils.parseEther(balances[0])
+      );
+    });
+
+    it('donations revert if not above minimum contribution', async function () {
+      const grantId: number = 0;
+      await mockERC20.mock.balanceOf.withArgs(donor.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.balanceOf.withArgs(grantPayee.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.transfer.withArgs(grantPayee.address, balances[0]).returns(true);
+
+      void expect(roundContract.connect(donor).donateToGrant(10, grantId)).to.be.revertedWith(
```

Comment:
These should all be `await` instead of `void`. With `void`, if the expect matcher would fail, you'll get an error logged to the console but since it's not `await`ed the test will still pass (even though it shouldn't)